src/psyclone/tests/psyir/transformations/intrinsics/matmul2code_trans_test.py

```diff
@@ -627,9 +626,54 @@ def test_validate_matmat_with_slices_on_rhs(fortran_reader):
     assign = psyir.walk(Assignment)[0]
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(assign.rhs)
-    assert (f"To use {trans.name} on matmul, each range on the result "
-            f"variable 'result' must be a full range but found "
-            f"result(2:4,2:5)" in str(excinfo.value))
+    assert ("Transformation Error: To use Matmul2CodeTrans on matmul, each "
+            "Range index of the argument 'result(2:4,2:5)' must be "
+            "a full range but found non full range "
+            "at position 0." in str(excinfo.value))
+
+
+def test_validate_res_too_many_full_ranges(fortran_reader):
+    '''
+    Check that the validate method refuses matrix-matrix operations with
+    too many full ranges in its lhs.
+
+    '''
+    psyir = fortran_reader.psyir_from_source(
+        "subroutine my_sub()\n"
+        "  real, dimension(2,6) :: jac\n"
+        "  real, dimension(6,3) :: jac_inv\n"
+        "  real, dimension(10,10,10) :: result\n"
+        "  result(:,:,:) = matmul(jac(:,:), jac_inv(:,:))\n"
+        "end subroutine my_sub\n")
+    trans = Matmul2CodeTrans()
+    assign = psyir.walk(Assignment)[0]
+    with pytest.raises(TransformationError) as excinfo:
+        trans.validate(assign.rhs)
+    assert ("Transformation Error: To use Matmul2CodeTrans on matmul, "
+            "no more than two indices of the argument 'result(:,:,:)' "
```

Comment:
s/argument/result variable/ perhaps?

Reply:
Addressed in 6fb1e5a1ffda53c4a956266e070b3a161dda58d0, I just call them all arrays now in that part of the validation.